api/get_rbg_hls_link.php

```diff
@@ -1,20 +1,26 @@
 <?php
 include 'simple_html_dom.php';
-include_once 'CacheAccess.php';
 
 function get_rbg_hls_link(string $url): ?string {
-    $html = CacheAccess::getHtml($url, 60 * 60 * 24); // this site should be static, therefore we cache it longer.
+    $CACHEKEY = "get_rbg_hls_link_" . $url;
+    $CACHETTL = 60 * 60 * 24; // One day
+    if (apcu_exists($CACHEKEY)) {
+        return apcu_fetch($CACHEKEY);
+    }
+
+    $html = file_get_html($url);
 
     if ($html) {
         $matches = [];
 
         preg_match('/MMstartStream.*\'(.+)\'/', $html, $matches);
 
         if (sizeof($matches) >= 1) {
+            apcu_add($CACHEKEY, $matches[0], $CACHETTL);
             return $matches[1];
         }
     }
-
+    apcu_add($CACHEKEY, null, $CACHETTL);
```

Comment:
Add a comment please: "// Cache empty results (as null) as well, so this edge case doesn't overload the source"